morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -256,6 +263,41 @@ public void ensureCaps() {
     }
 
     @Override
+    public void enableDocumentValidation() {
+        for (final MappedClass mc : mapper.getMappedClasses()) {
+            process(mc, (Validation) mc.getAnnotation(Validation.class));
+        }
+    }
+
+    void process(final MappedClass mc, final Validation validation) {
+        if (validation != null) {
+            String collectionName = mc.getCollectionName();
+            CommandResult result = getDB()
+                .command(new BasicDBObject("collMod", collectionName)
+                             .append("validator", parse(validation.value()))
+                             .append("validationLevel", validation.level().getValue())
+                             .append("validationAction", validation.action().getValue())
+                        );
+
+            if (!result.ok()) {
+                if (result.getInt("code") == 26) {
+                    try {
+                        ValidationOptions options = new ValidationOptions()
+                            .validator(parse(validation.value()))
+                            .validationLevel(validation.level())
+                            .validationAction(validation.action());
+                        getDatabase().createCollection(collectionName, new CreateCollectionOptions().validationOptions(options));
+                    } catch (JsonParseException e) {
+                        LOG.warning(format("Could not parse validator for '%s:'  %s", collectionName, e.getMessage()));
+                    }
+                } else {
+                    LOG.warning(format("Could not add document validation on '%s:'  %s", collectionName, result.getErrorMessage()));
```

Comment:
This is going to create inconsistent results.  If collMod fails for some real reason (like not authorized to modify the collection), it will just log a warning.  But if create fails for the same reason, it will throw an exception.

What does index creation do if there's an error creating the index?  I suggest we make the behavior here consistent with whatever that is.